hazelcast-client/src/test/java/com/hazelcast/client/ClientConnectionTest.java

```diff
@@ -98,4 +105,35 @@ public void testMemberConnectionOrder() {
         Collection<Client> connectedClients2 = server2.getClientService().getConnectedClients();
         assertEquals(connectedClients2.size(), 0);
     }
+
+    @Test
+    public void testConnectionRemovedOnce() {
+        HazelcastInstance server = Hazelcast.newHazelcastInstance();
+        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        HazelcastClientInstanceImpl clientImpl = ClientTestUtil.getHazelcastClientInstanceImpl(client);
+        ClientConnectionManager connectionManager = clientImpl.getConnectionManager();
+
+
+        final AtomicInteger count = new AtomicInteger();
+
+        connectionManager.addConnectionListener(new ConnectionListener() {
+            @Override
+            public void connectionAdded(Connection connection) {
+
+            }
+
+            @Override
+            public void connectionRemoved(Connection connection) {
+                count.incrementAndGet();
+            }
+        });
+
+        final Address serverAddress = new Address(server.getCluster().getLocalMember().getSocketAddress());
+        final Connection connectionToServer = connectionManager.getConnection(serverAddress);
```

Comment:
I understand your point and I agree with most of it! 

My goal is to make the test easier to read. Currently there are two lines, one of them is quite long and not exactly easy to read. Neither line is important in order to follow a logic of the test. It can be treated as "helper" or "utility" code or perhaps even as a "noise". 

I believe factoring this code out into its own method will make the test more concise - the test itself just requires a connection, it's not important how the connection is obtained. 

Anyway, it's not terribly important, it's just 2 lines. We could merge it as it is, but I believe it would make the code better.